### PR TITLE
[MULTI]Data affinity remote context and blobs

### PR DESCRIPTION
**Pull request by myshevts**
The request is checked for the input RemoteBlobs (as an additional pipeline step) and (since RemoteBlobs are NOT compatible between devices) set the "preferred" device accordingly

### CODE REVIEW

**Review by apankratovantonp — COMMENTED** [review]
Please merge after https://github.com/openvinotoolkit/openvino/pull/3300

File: inference-engine/tests/functional/plugin/shared/include/multi/multi_remote_blob_tests.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "multi/multi_helpers.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+
+TEST_P(MultiDevice_SupportTest, canCreateContextThenRequestThenBlobsAndInfer) {
+    InferenceEngine::CNNNetwork net;
+    net = CNNNetwork(fn_ptr);
+    net.getInputsInfo().begin()->second->setLayout(Layout::NCHW);
+    net.getInputsInfo().begin()->second->setPrecision(Precision::U8);
+
+    auto ie = InferenceEngine::Core();
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/include/multi/multi_remote_blob_tests.hpp:16]: minor comment, you can use core taken from `plugin cache` there

**Review by iefode — APPROVED** [review]

**Review by rfsaliev — APPROVED** [review]

**Comment by myshevts** [comment]
> Please merge after #3300
It is now based both on the PR-3300 and PR-3286.
@apankratovantonp pls review.

**Comment by myshevts** [comment]
Merging to unlock further work on general IE exception sanitizing (some items already were required for this PR as well).
Any possible further comments, e.g. from @apankratovantonp are still welcome.

**Pull request merged**